### ISSUE TITLE
Allow CREATE SUBSCRIPTION command without needing superuser privilege

### ISSUE THREAD

**Issue by burggraf** [supabase/supautils#103]
Supabase currently supports:
`CREATE PUBLICATION my_publication FOR ALL TABLES;`

but does not support:
`CREATE SUBSCRIPTION my_subscription CONNECTION '<connection_string>' PUBLICATION my_publication;`

The error given is:
`ERROR:  42501: must be superuser to create subscriptions`

In order to be able to use logical replication to replicate individual tables (or all tables in a schema) between Supabase projects, `CREATE SUBSCRIPTION` should be executable by the `postgres` user without `supseruser` privileges, in parity with the `CREATE PUBLICATION` command.

**Comment by steve-chavez** [comment]
This will come for free once pg 17 is supported.

**Comment by t0mdicks0n** [comment]
I followed the same guide on how to setup replication on Supabase but superuser is needed so the guide is obsolete? At least remove that page on the docs almost 1 year after users in this thread notified you about it?

[https://supabase.com/docs/guides/database/postgres/setup-replication-external](https://supabase.com/docs/guides/database/postgres/setup-replication-external)

**Comment by encima** [comment]
Hey folks,

The guide there is for replicating to an `external` system so the subscription is created on the target database. Creating publications etc are supported without superuser permissions

**Comment by Lanznx** [comment]
any update?

**Comment by encima** [comment]
Yep, this is possible for the `postgres` user as of 17.4 and is rolling out
across regions currently
